test/unit/math/prim/mat/prob/vector_rng_test_helper.hpp

```diff
@@ -177,48 +253,151 @@ void check_dist_throws(T_generate_samples generate_samples,
+  check_dist_throws<VectorXd, VectorXd, VectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<VectorXd, RowVectorXd, VectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, double, VectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, std::vector<double>, VectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, VectorXd, VectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, RowVectorXd, VectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+
+  check_dist_throws<double, double, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<double, std::vector<double>, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<double, VectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<double, RowVectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<std::vector<double>, double, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<std::vector<double>, std::vector<double>, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<std::vector<double>, VectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<std::vector<double>, RowVectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<VectorXd, double, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<VectorXd, std::vector<double>, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<VectorXd, VectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<VectorXd, RowVectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, double, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, std::vector<double>, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, VectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
+  check_dist_throws<RowVectorXd, RowVectorXd, RowVectorXd>
+    (generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3);
```

Comment:
Whenever you see this much redundant code, you want to ask yourself if it's really necessary.  

My first inclination here would be to wrap up the arguments `(generate_samples, good_p1, bad_p1, good_p2, bad_p2, good_p3, bad_p3)` into a structure.  Then it's just `check_dist_throws<T1, T2, T3>(s)` where `s` is the holding structure.  That also simplifies the argument declaration for `check_dist_throws`.   I think you should make this change.

I think you also want to be checking `int`---it can sometimes cause problems you don't see with `double` with argument ambiguity resolution.  I also think you should make this change.

My second inclination would be to use typelists to iterate over all the types.  It's a template programming technique worth learning and a little recursion will deal with all the permutations in a way that's probably easier to validate is correct.  The problem with these big lists is that it's very hard to check that they're complete.  And they're hard to maintain when you add a new type.  This change is optional, but I'd encourage you to at least check out the template technique and start looking over your shoulder every time you think about creating long lists like this.